src/item_search.cpp

```diff
@@ -8,34 +8,62 @@
 
 std::pair<std::string, std::string> get_both( const std::string &a );
 std::function<bool( const item & )>
-item_filter_from_string(  std::string filter )
+item_filter_from_string( std::string filter )
 {
     if( filter.empty() ) {
         // Variable without name prevents unused parameter warning
-        return [](const item&) {
+        return []( const item & ) {
             return true;
         };
     }
-    if( filter.find( "," ) != std::string::npos ){
-        std::vector<std::function<bool( const item& )> > functions;
+    
+    // remove curly braces (they only get in the way)
+    filter.erase( std::remove( filter.begin(), filter.end(), '{') );
+    filter.erase( std::remove( filter.begin(), filter.end(), '}') );
+    if( filter.find( "," ) != std::string::npos ) {
+        // functions which only one of which must return true
+        std::vector<std::function<bool( const item & )> > functions;
+        // Functions that must all return true
+        std::vector<std::function<bool( const item & )> > inv_functions;
         size_t comma;
         while( ( comma = filter.find( "," ) ) != std::string::npos ) {
-            const auto& current_filter = filter.substr( 0, comma );
-            if( current_filter.empty() ){
-                continue;
+            const auto &current_filter = trim( filter.substr( 0, comma ) );
+            if( !current_filter.empty() ) {
+                auto current_func = item_filter_from_string( current_filter );
+                if( current_filter[0] == '-' ) {
+                    inv_functions.push_back( current_func );
+                } else {
+                    functions.push_back( current_func );
+                }
             }
-            auto current_func = item_filter_from_string( current_filter );
-            functions.push_back( current_func );
             filter = trim( filter.substr( comma + 1 ) );
         }
-        if( !filter.empty() ){
-            functions.push_back( item_filter_from_string( filter ) );
+        if( !filter.empty() ) {
+            std::cout << filter << " " << std::endl;
+            if( filter[0] != '-' ) {
+                functions.push_back( item_filter_from_string( filter ) );
+            }else {
+                inv_functions.push_back( item_filter_from_string( filter) );
+            }
         }
-        return [functions](const item& i) {
-            return std::any_of( functions.begin(), functions.end(),
-            [i](std::function<bool(const item&)> f){
+        return [functions, inv_functions]( const item & i ) {
+            bool p_result = std::any_of( functions.begin(), functions.end(),
+            [i]( std::function<bool( const item & )> f ) {
```

Comment:
`Item` should be captured by reference here. Also, I would rename it to `it` (pun unintended).